AUTHENTICATION.md

```diff
@@ -343,3 +344,49 @@ spec:
     source: InjectedIdentity
 EOF
 ```
+
+## Using `assumeRoleWithWebIdentity`
+
+`provider-aws` will be configured to connect to the aws account in `RoleARN` and request
+a session for `RoleARN` using it's `InjectedIdentity` 
+
+This is most useful when "sts chaining" (see [Using `assumeRole`](#using-assumerole)) 
```

Comment:
We should document the requirement for IdP and trust.

### Requirements:
1.  Create a IdP in target AWS accounts. Example OIDC provider for EKS:
```bash
aws iam create-open-id-connect-provider -url https://oidc.eks.us-west-2.amazonaws.com/id/123456789 --thumbprint-list 9E99A48A9960B14926BB7F3B02E22DA2B0AB7280 --client-id-list sts.amazonaws.com
{
    "OpenIDConnectProviderArn": "arn:aws:iam::1234:oidc-provider/oidc.eks.us-west-2.amazonaws.com/id/123456789"
}

```
Save the return ARN for next step.

2. Create a role for the AWS provider SA to assume. 
```bash 
cat > trust.yaml <<EOF
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Effect": "Allow",
      "Principal": {
        "Federated": "${ARN_FROM_ABOVE}"
      },
      "Action": "sts:AssumeRoleWithWebIdentity",
      "Condition": {
        "StringLike": {
          "${OIDC_PROVIDER}:sub": "system:serviceaccount:${SERVICE_ACCOUNT_NAMESPACE}:provider-aws-*"
        }
      }
    }
  ]
}
EOF

aws iam create-role \
    --role-name "${IAM_ROLE_NAME}" \
    --assume-role-policy-document file://trust.json \
    --description "IAM role for provider-aws"
```
3. Verify. The command below assumes the token file is available at `/tmp/token` 
```bash
aws sts assume-role-with-web-identity --role-arn arn:aws:iam::123:role/${IAM_ROLE_NAME} --role-session-name test --web-identity-token `cat /tmp/token`
```

Reply:
We have also an example in upbound platform ref configure it directly with composition https://github.com/upbound/platform-ref-aws/pull/56 - just for example ;)

Reply:
Yeah that's a good idea. We just need to make sure we document the need for OIDC provider in each target account and trust relationships. These are created by eksctl for the account that the cluster lives in, but not target accounts.

Reply:
FYI I am still working up this change, however it should not block any functional review.

Reply:
@nabuskey I added pretty extensive docs that I think cover the requirements. @haarchri I decided to avoid a circular dependency on crossplane being configured to target the account in question by simpy using aws cli commands. Let me know what you think.